src/Scripting/CoreTestUtilities/RedirectedOutput.cs

```diff
@@ -2,16 +2,21 @@
 
 using System;
 using System.IO;
+using System.Threading;
 
 namespace Microsoft.CodeAnalysis.Scripting
 {
     public sealed class OutputRedirect : IDisposable
     {
+        private static readonly object s_guard = new object();
+
         private readonly TextWriter _oldOut;
         private readonly StringWriter _newOut;
 
         public OutputRedirect(IFormatProvider formatProvider)
         {
+            Monitor.Enter(s_guard);
```

Comment:
I'm guessing the Monitor is used because the release happens when the object is disposed. I also feel like there are better ways to do this than using a monitor to lock over the lifetime of the object.